VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -205,5 +206,62 @@ public static bool IsNameofInvocation(this InvocationExpressionSyntax invocation
 
             return identifier != null && identifier.Identifier.ValueText == "nameof";
         }
+
+        public static List<T> SyntaxNodeOfType<T>(this IEnumerable<SyntaxNode> enumerable, SyntaxKind kind) where T : SyntaxNode
```

Comment:
This one isn't necessary since `OfType<T>` doesn't allocated a lambda anyway and I believe this implementation is largely the same.